transifex/utils.py

```diff
@@ -340,6 +372,49 @@ def merge_pr(self):
         )
         raise RuntimeError('Pending status checks or other failure prevented this PR from being automatically merged')
 
+    def _attempt_merge(self):
+        try:
+            self.pr.merge(merge_method=self.merge_method)
+            logger.info('Merged [%s/#%d].', self.name, self.pr.number)
+            return True
+        except GithubException as e:
+            logger.error(
+                'Failed to merge [%s/#%d], because of the exception [%s]',
+                self.name, self.pr.number, e,
+            )
+            raise RuntimeError('Failed to merge.')
+
+    def _get_tests_combined_status(self):
+        """ Returns combined status of pr tests """
+        commit = self.pr.get_commits().reversed[0]
+
+        total_statuses = commit.get_statuses().totalCount
+        if total_statuses > 0:
+            combined_status = commit.get_combined_status().state
+            logger.info(f'Combined status for {self.name}/#{self.pr.number}: {combined_status}')
+            if combined_status in [PENDING, FAILURE]:
+                return combined_status
+
+        # Check runs are not included in statuses, iterate over these before determining
```

Comment:
what are check runs?

Reply:
[commit statuses](https://docs.github.com/en/rest/commits/statuses) are generated by external services (e.g. codecov), while check runs use the [GH Checks API](https://docs.github.com/en/rest/checks/runs) (so basically all github-actions tests). `get_combined_status` only takes into account the former, unfortunately

Reply:
[commit statuses](https://docs.github.com/en/rest/commits/statuses) are generated by external services (e.g. codecov), while check runs use the [GH Checks API](https://docs.github.com/en/rest/checks/runs) (so basically all github-actions tests). `get_combined_status` only takes into account the former, unfortunately

Reply:
on that note, I think our translation PRs were merging without waiting for CI before 😶